Update `gh variable get` to use repo host

andyfeller opened the pull request:
Fixes #9274

This change addresses a bug where repository and repository environment variables are not being retrieved from the appropriate host, causing GHES users to fail without specifying `GH_HOST` to force `gh variable set` to do the right thing.

In addition, the tests for this command have been updated to check both github.com and GHES variations to ensure this works.

cc: @yukha-dw

<!--
  Thank you for contributing to GitHub CLI!
  To reference an open issue, please write this in your description: `Fixes #NUMBER`
-->

andyfeller committed:
Update `gh variable get` to use repo host

Closes #9274

This change addresses a bug where repository and repository environment variables are not being retrieved from the appropriate host, causing GHES users to fail without specifying `GH_HOST` to force `gh variable set` to do the right thing.

In addition, the tests for this command have been updated to check both github.com and GHES variations to ensure this works.

jtmcg commented on pkg/httpmock/stub.go:
```diff
@@ -123,6 +123,15 @@ func StringResponse(body string) Responder {
 
+func WithHost(matcher Matcher, host string) Matcher {
+	return func(req *http.Request) bool {
+		if !strings.EqualFold(req.Host, host) {
```
TIL: `strings.EqualFold` matches with case insensitivity

jtmcg reviewed (commented):
Have validated that this works locally and the code looks good! Once we decide what to do about `gh variable delete` and `gh secret delete` I'll approve. I vote a follow up PR. Will have to keep the issue open, though, if we go that route, until that PR is up.

Referring to this [slack comment](https://github.slack.com/archives/CLLG3RMAR/p1722615171112239?thread_ts=1722609825.891839&cid=CLLG3RMAR)

andyfeller committed:
Fix host handling in variable and secret delete

This change updates `gh variable delete` and `gh secret delete` to use the host associated with the repository rather than the default host.  Along with these changes is minor refactoring of the tests to ensure appropriate Dotcom vs GHES targeting works as expected.

Minor edit to `gh variable get` testing to simplify some conditional logic in test setup.

jtmcg reviewed (approved):
Changes look good and I've tested the 3 commands. :shipit:

Donald135 commented on pkg/cmd/secret/delete/delete.go:
```diff
@@ -105,25 +105,28 @@ func removeRun(opts *DeleteOptions) error {
 	case shared.Environment:
 		path = fmt.Sprintf("repos/%s/environments/%s/secrets/%s", ghrepo.FullName(baseRepo), envName, opts.SecretName)
+		host = baseRepo.RepoHost()
 	case shared.User:
```
4c088951944082eb4c474097edb1861ed07d6c856

Donald135 commented on pkg/cmd/secret/delete/delete_test.go:
```diff
@@ -121,41 +122,71 @@ func TestNewCmdDelete(t *testing.T) {
 func Test_removeRun_repo(t *testing.T) {
 	tests := []struct {
-		name     string
-		opts     *DeleteOptions
```
pkg/cmd/secret/delete/delete_test.go

Donald135 commented on pkg/cmd/secret/delete/delete.go:
pkg/cmd/secret/delete/delete.go

Donald135 commented on pkg/cmd/secret/delete/delete.go:
```diff
@@ -105,25 +105,28 @@ func removeRun(opts *DeleteOptions) error {
 		}
 	}
 
```
gh.secret/delete.go
